src/io-methods/azure.js

```diff
@@ -1,38 +1,52 @@
-const azure = require("azure-storage");
+const { BlobServiceClient } = require("@azure/storage-blob");
 let blobService;
 
-function getBlobService() {
-  blobService = blobService || azure.createBlobService();
+const AZURE_STORAGE_CONNECTION_STRING =
+  process.env.AZURE_STORAGE_CONNECTION_STRING;
+
+async function getBlobService() {
+  blobService =
+    blobService ||
+    (await BlobServiceClient.fromConnectionString(
+      AZURE_STORAGE_CONNECTION_STRING
+    ));
   return blobService;
 }
 
-exports.readManifest = function (target) {
-  const blobService = getBlobService();
-
-  return new Promise(function (resolve, reject) {
-    blobService.getBlobToText(
-      target.azureContainer,
-      target.azureBlob,
-      function (error, response) {
-        if (error) reject(error);
-        resolve(response);
-      }
-    );
+async function streamToString(readableStream) {
+  return new Promise((resolve, reject) => {
+    const chunks = [];
+    readableStream.on("data", (data) => {
+      chunks.push(data.toString());
+    });
+    readableStream.on("end", () => {
+      resolve(chunks.join(""));
+    });
+    readableStream.on("error", reject);
   });
-};
+}
+
+// Reference: https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage/storage-blob#download-a-blob-and-convert-it-to-a-string-nodejs
+exports.readManifest = async function (target) {
+  const blobService = await getBlobService();
+  const containerClient = blobService.getContainerClient(target.azureContainer);
+  const blobClient = containerClient.getBlobClient(target.azureBlob);
 
-exports.writeManifest = function (target, content) {
-  const blobService = getBlobService();
+  const downloadBlockBlobResponse = await blobClient.download();
+  const response = await streamToString(
+    downloadBlockBlobResponse.readableStreamBody
+  );
+  return response;
+};
 
-  return new Promise(function (resolve, reject) {
-    blobService.createBlockBlobFromText(
-      target.azureContainer,
-      target.azureBlob,
-      content,
-      function (error, response) {
-        if (error) reject(error);
-        else resolve();
-      }
-    );
+exports.writeManifest = async function (target, content) {
+  const blobService = await getBlobService();
+  const containerClient = blobService.getContainerClient(target.azureContainer);
+  const blockBlobClient = containerClient.getBlockBlobClient(target.azureBlob);
+  return await blockBlobClient.upload(content, content.length, {
+    blobHTTPHeaders: {
+      blobCacheControl: "public, must-revalidate, max-age=0",
+      blobContentType: "application/json",
```

Comment:
```suggestion
      blobContentType: "application/importmap+json",
```

 See https://github.com/single-spa/import-map-deployer/issues/51